# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MIROps.yaml
# Commit: 28c38d556fb8
# Full Hash: 28c38d556fb86f1a936d9f6a8627705230df895a
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2021-12-09 09:32:28
# Regressor Bug: 1713579
# File Overlap Count: 1
# Description:
#   Bug 1713579: Extract shape-guard folding into separate optimization r=jandem
#   
#   This patch takes the code from `MGuardShape::foldsTo` and splits it out into its own optimization. Besides jitspew, the main change is adding support for MNewPlainObject (which didn't exist when this code was first written.)
#   
#   We use `dependency()` to find the most recent store. It is possible for a node's dependency to be in a dead block. In `ValueNumberer::visitDefinition`, we check for this case; I used the same approach here.
# ==============================================================================

diff -r a09f76aa6703 -r 28c38d556fb8 js/src/jit/MIROps.yaml
--- a/js/src/jit/MIROps.yaml	Wed Dec 08 21:19:47 2021 +0000
+++ b/js/src/jit/MIROps.yaml	Wed Dec 08 21:19:49 2021 +0000
@@ -1321,7 +1321,6 @@
   guard: true
   movable: true
   congruent_to: custom
-  folds_to: custom
   alias_set: custom
 
 - name: GuardProto